### PR TITLE
WIP: Shift/Arrow key cycle + Help Prompt + Indent fix

### PR DESCRIPTION
**Pull request by DecathectZero**

### CODE REVIEW

File: app/assets/javascripts/Grader/marking.js [hunk 1]
```diff
@@ -260,16 +268,48 @@ function activeCriteria($next_criteria){
...
+        $next_criteria.find('.criterion_expand').click();
+      }
+    }
+  }
+}
+
+// Hide the expansion of the current active-criterion
+function hideActiveCriterion() {
+  $active = $('.active-criterion');
+  if($active.hasClass('expanded')){
```
Comment (david-yz-liu) [inline app/assets/javascripts/Grader/marking.js]: Please make sure whitespace guidelines are followed throughout.

File: app/views/results/marker/_keybinding.js.erb [hunk 2]
```diff
@@ -8,12 +8,26 @@ Mousetrap.bind('>', function() {
   $('#student_selector>.right>a')[0].click();
 });
 
+// Go to the previous criterion with shift + up
+Mousetrap.bindGlobal('shift+up', function(e) {
```
Comment (david-yz-liu) [inline app/views/results/marker/_keybinding.js.erb]: I'm not sure `bindGlobal` is a good idea here. There are other text boxes where I could see this shortcut conflicting. I'm assuming you put this in for the flexible criteria inputs - apparently if you add `class="mousetrap"` to the input, all bindings will go through. I think this is okay for the inputs?

File: app/views/results/marker/_keybinding.js.erb [hunk 3]
```diff
@@ -8,12 +8,26 @@ Mousetrap.bind('>', function() {
...
+  e.preventDefault();
+  prevCriterion();
+  return false;
+});
+
+// Go to the next criterion with shift + down
+Mousetrap.bindGlobal('shift+down', function(e) {
+  e.preventDefault();
+  nextCriterion()
```
Comment (david-yz-liu) [inline app/views/results/marker/_keybinding.js.erb]: Please insert a semicolon here.

File: app/views/results/marker/_keybinding.js.erb (hunk 2 above)
Comment (DecathectZero) [inline app/views/results/marker/_keybinding.js.erb]: yeah that would work much better then.

**Comment by DecathectZero** [comment]
The problem with the radio buttons not changing with arrow keys still persists.

File: app/views/results/edit.html.erb [hunk 4]
```diff
@@ -134,6 +135,39 @@
                        can_download: true } %>
 
 </aside>
+<aside class='dialog' id='help_dialog'>
+  <h1>Keyboard Shortcut Help</h1>
```
Comment (david-yz-liu) [inline app/views/results/edit.html.erb]: Please internationalize these strings.

File: app/views/results/edit.html.erb [hunk 5]
```diff
@@ -134,6 +135,39 @@
...
+  <table class="help_table">
+    <tr>
+      <td>Shift + ></td>
```
Comment (david-yz-liu) [inline app/views/results/edit.html.erb]: Please use [`kbd`](https://developer.mozilla.org/en/docs/Web/HTML/Element/kbd) HTML elements for these. You can add some CSS to style them if you want.

File: app/assets/stylesheets/results.scss [hunk 6]
```diff
@@ -122,3 +122,15 @@
 input:focus {
   border: 5px solid $blue;
 }
+
+#help_dialog table{
```
Comment (david-yz-liu) [inline app/assets/stylesheets/results.scss]: Please use spaces here as well.

File: app/views/results/marker/_keybinding.js.erb [hunk 7]
```diff
@@ -1,3 +1,8 @@
+// Open the keyboard shortcuts help modal
+Mousetrap.bind('?', function () {
```
Comment (david-yz-liu) [inline app/views/results/marker/_keybinding.js.erb]: Pass `modal_help.open` as the callback (no need for the anonymous function).

**Pull request merged**